tests/mutual-tls.bats

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bats
+
+setup() {
+    load ../helpers/helpers.sh
+    wait_pods
+}
+
+setup_file() {
+    load ../helpers/helpers.sh
+    kubectl get cm -n $NAMESPACE mtlscm &>/dev/null || kubectl create cm -n $NAMESPACE mtlscm --from-file="client-ca.crt=$RESOURCES_DIR/mtls/rootCA.crt"
+}
+
+teardown_file() {
+    load ../helpers/helpers.sh
+    helmer reset kubewarden-defaults
+    helmer reset kubewarden-controller
+}
+
+# Skip tests if MTLS is disabled
+[[ "${MTLS:-}" =~ ^(false|0)?$ ]] && skip "Mutual TLS is disabled"
+
+@test "[Mutual TLS] Enable mTLS" {
+    helmer set kubewarden-controller --set mTLS.enable=true --set mTLS.configMapName=mtlscm
+
+    kubectl get nodes -l node-role.kubernetes.io/control-plane -o yaml | grep -F "admission-control-config-file"
+    kubectl logs -n kubewarden -l kubewarden/policy-server=default | grep -E "certs: Loaded client CA certificates client_ca_certs_added=[1-9]"
+
+    # Run more checks? What to check for?
```

Comment:
Ideally I would like to:
- Check that TLS auth by certs (in addition to the usual bearer token) is enabled on the API server for the webhooks. I suppose this means hitting the IP of the node where the api-server is, under /api and port 443 (6443 is exposed externally).
- Check that TLS auth is enabled on our side. I think with something like:
  ```
  kubectl port-forward -n kubewarden service/policy-server-default 3000:8443
  curl https://localhost:3000/validate --verbose # should fail because of missing certs at least
  ```
- Trigger a policy run by creating/updating a pod, and see that the policy-server rejects
- Trigger an audit-scanner run and see that there's no errors on its log (or on the resulting policyreports)

Reply:
Another option is to enable mTLS by default and run all the tests as usual. Thus, we test mTLS in all possible scenarios

Reply:
@jvanz I will enable mTLS fully but only in some scenarios (to check we don't actually require it by mistake), see https://github.com/kubewarden/helm-charts/pull/647 for details

@viccuad I will merge to have something already running on helm charts nightly jobs and will look into your suggestions in follow-up PR